pkg/notifier/notifier.go

```diff
@@ -0,0 +1,176 @@
+package notifier
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/pingcap/errors"
+	"go.uber.org/atomic"
+
+	"github.com/hanfei1991/microcosm/pkg/containers"
+)
+
+type receiverID = int64
+
+// Notifier is the sending endpoint of a single-producer-multiple-consumer
+// notification mechanism.
+type Notifier[T any] struct {
+	receivers sync.Map // receiverID -> *Receiver[T]
+	nextID    atomic.Int64
+
+	queue *containers.SliceQueue[T]
+
+	closeCh       chan struct{}
+	synchronizeCh chan struct{}
```

Comment:
Can you add more comment on the `synchronizeCh`?
Can't the point.

Reply:
Get the point.
`synchronizeCh` is to achieve async-close the receiver `C` safely.
Pls add some comment if possible. Thanks.